src/item/find_descr.py

```diff
@@ -24,30 +23,36 @@ def choose_best_result(res_left: SearchResult, res_right: SearchResult) -> Searc
     elif res_left.success and res_right.success:
         return res_left if res_left.matches[0].score > res_right.matches[0].score else res_right
     else:
-        return None
+        return SearchResult(success=False)
 
 
 def find_descr(img: np.ndarray, anchor: tuple[int, int]) -> tuple[bool, ItemRarity, np.ndarray, tuple[int, int, int, int]]:
-    item_descr_width = Config().ui_offsets["item_descr_width"]
-    item_descr_pad = Config().ui_offsets["item_descr_pad"]
-    _, window_height = Config().ui_pos["window_dimensions"]
+    item_descr_width = ResManager().offsets.item_descr_width
+    item_descr_pad = ResManager().offsets.item_descr_pad
+    _, window_height = ResManager().pos.window_dimensions
 
     refs = list(map_template_rarity.keys())
     res_left = None
     res_right = None
 
-    roi_left = copy(Config().ui_roi["rel_descr_search_left"])
+    roi_left = ResManager().roi.rel_descr_search_left
     roi_left[0] += anchor[0]
-    ok, roi_left = fit_roi_to_window_size(roi_left, Config().ui_pos["window_dimensions"])
+    ok, roi_left = fit_roi_to_window_size(roi_left, ResManager().pos.window_dimensions)
     if ok:
         res_left = search(ref=refs, inp_img=img, roi=roi_left, threshold=0.8, mode="best")
 
-    roi_right = copy(Config().ui_roi["rel_descr_search_right"])
+    roi_right = ResManager().roi.rel_descr_search_right
     roi_right[0] += anchor[0]
-    ok, roi_right = fit_roi_to_window_size(roi_right, Config().ui_pos["window_dimensions"])
+    ok, roi_right = fit_roi_to_window_size(roi_right, ResManager().pos.window_dimensions)
     if ok:
         res_right = search(ref=refs, inp_img=img, roi=roi_right, threshold=0.8, mode="best")
 
+    # TODO why is this happening?!
```

Comment:
@aeon0 I have triple checked my code and can't find the error. Can you help please?

Comparing this branch and main, debugging shows the same data and type. But for some reason both res are always None using this branch.

Reply:
Found it